### ISSUE TITLE
Flathub version produces jpg's with size of 0 bytes

### ISSUE THREAD

**Issue by axredneck** [Huluti/ImCompressor#32]
So currently it's useless for jpg's, or even harmful when saving to the same file.

**Comment by Huluti** [comment]
You're right... I've become aware of this problem a few days ago. However, I don't find any reason for this problem as this does not occur in dev mode, only on flathub...
I will keep you posted very quickly with a solution I hope.

**Comment by ramcq** [comment]
To me this sounds more like a problem with file access from portals - maybe some incompatibility between a file chooser (Gtk/Qt/etc) which supports portals, and then using that file with a library that does not, and so is writing into the wrong location? Can you change/remove the issue if you use overrides like --filesystem=home?

**Comment by axredneck** [comment]
@ramcq, it processes pngs from the same folder without problems

**Comment by ramcq** [comment]
Right, but libpng could be handling its files differently to libjpeg, for example. It's not 100% guaranteed that code paths are the same for different formats.

**Comment by Huluti** [comment]
@ramcq thank's for helping us! ""--filesystem=home" is already in the manifest. Do you mean an other thing?

**Comment by Huluti** [comment]
@ramcq do you have any clue?

**Comment by barthalion** [comment]
Correct me if I'm wrong, but at first glance, ImCompressor seems to use `jpegoptim`:

```
imcompressor/imcompressor/compressor.py:        jpegoptim = 'jpegoptim --max={} -o -f --stdout "{}" > "{}"'
imcompressor/imcompressor/compressor.py:        jpegoptim2 = 'jpegoptim -o -f --stdout "{}" > "{}"'
```

However running `find /app -name jpegoptim` inside the app finds nothing.

**Comment by Huluti** [comment]
@barthalion You're right, I use `jpegoptim`.

That's strange because jpegoptim is in the file : `com.github.huluti.ImCompressor.json`.

**Comment by barthalion** [comment]
Well, yes, in your repository. You also need to push it to https://github.com/flathub/com.github.huluti.ImCompressor.

**Comment by Huluti** [comment]
Oh shame ... it was so simple. Sincerely sorry for the inconvenience. I take care of it very quickly ... totally my fault.

**Comment by Huluti** [comment]
Should be fixed in the latest flathub release: https://github.com/flathub/com.github.huluti.ImCompressor/commit/7a9224b54dd884ec6406fbbc9cdfe9312f6a4a81

**Comment by axredneck** [comment]
@Huluti , not fixed, still 0 byte files

**Comment by Huluti** [comment]
> @Huluti , not fixed, still 0 byte files

have you update the app?

**Comment by axredneck** [comment]
@Huluti 
```
redneck@pylesos ~> flatpak update
Looking for updates…
Info: org.gnome.Sdk is end-of-life, with reason: The GNOME 3.32 runtime is no longer supported as of 11th March 2020. Please ask your application developer to migrate to a supported platform.
Info: org.gnome.Sdk.Locale is end-of-life, with reason: The GNOME 3.32 runtime is no longer supported as of 11th March 2020. Please ask your application developer to migrate to a supported platform.
Info: org.gnome.Platform is end-of-life, with reason: The GNOME 3.32 runtime is no longer supported as of 11th March 2020. Please ask your application developer to migrate to a supported platform.
Info: org.gnome.Platform.Locale is end-of-life, with reason: The GNOME 3.32 runtime is no longer supported as of 11th March 2020. Please ask your application developer to migrate to a supported platform.
Nothing to do.
```

**Comment by barthalion** [comment]
Please double check with `flatpak info` if it's up to date. It works for me with the latest build, which is:

```
ImCompressor - Simple & useful image compressor

          ID: com.github.huluti.ImCompressor
         Ref: app/com.github.huluti.ImCompressor/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 0.8
     License: GPL-3.0-or-later
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 358,9 kB
     Runtime: org.gnome.Platform/x86_64/3.36
         Sdk: org.gnome.Sdk/x86_64/3.36

      Commit: 6b4362edc30562b77df478948095d615282c62d4d989dc8162d4f91533a03ae8
      Parent: 0a486847fe48b8e82b5d73650135a81bb44942a3baf7d6b1ca6d243e9622a3b6
     Subject: Fix deps and update to gnome sdk 3.36 (7a9224b5)
        Date: 2020-04-01 17:42:05 +0000
```

**Comment by axredneck** [comment]
```
redneck@pylesos ~ [1]> flatpak info com.github.huluti.ImCompressor

ImCompressor - Simple & useful image compressor

              ID: com.github.huluti.ImCompressor
          Ссылка: app/com.github.huluti.ImCompressor/x86_64/stable
     Архитектура: x86_64
           Ветка: stable
          Версия: 0.8
         License: GPL-3.0-or-later
        Источник: flathub
       Коллекция: org.flathub.Stable
       Установка: user
      Установлен: 358.9 kB
Среда исполнения: org.gnome.Platform/x86_64/3.36
             Sdk: org.gnome.Sdk/x86_64/3.36

          Коммит: 6b4362edc30562b77df478948095d615282c62d4d989dc8162d4f91533a03ae8
        Родитель: 0a486847fe48b8e82b5d73650135a81bb44942a3baf7d6b1ca6d243e9622a3b6
       Заголовок: Fix deps and update to gnome sdk 3.36 (7a9224b5)
             Дата 2020-04-01 17:42:05 +0000
```
Still an issue, still 0 bytes jpeg files.

**Comment by Huluti** [comment]
Once again you're right. I can reproduce the problem by disabling the "save in a new file" option... I continue to search for a solution.

**Comment by axredneck** [comment]
@Huluti why did you close the issue?

**Comment by Huluti** [comment]
@axredneck Sorry my commit have closed the issue automatically. I've just released ImCompressor 0.8.1 which should fixed the bug definitely. You just have to wait that it is published on flathub (~1h). Please feel free to reopen the issue if it doesn't work for you again. Thank you very much for your special help and thank you to @barthalion and @ramcq also ;)
